solutions/usaco-491.mdx

```diff
@@ -10,6 +10,71 @@ author: Qi Wang
 **Time Complexity:** $\mathcal{O}(N)$
 
 <LanguageSection>
+	
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+unordered_map<int, vector<int>> mp;
+vector<vector<int>> dist;
+
+void bfs(int s, int ix)
+{
+	queue<pair<int, int>> q;
+	q.push({s, 0});
+	dist[ix][s] = 0;
+	while (!q.empty())
+	{
+		pair<int, int> p = q.front();
+		q.pop();
+		for (auto n : mp[p.first])
+		{
+			if (dist[ix][n] != -1)
+			{ 
+				continue; 
+			}
+			dist[ix][n] = p.second + 1;
+			q.push({n, dist[ix][n]});
+		}
+	}
+}
+
+int main()
+{
+	int B, E, P, N, M; 
+    	cin >> B >> E >> P >> N >> M;
+
+    	dist.assign(4, vector<int>(N + 1, -1));
+    	for (int i = 0; i < M; i++)
+    	{
+		int u, v;
+        	cin >> u >> v;
+        	u -= 1;
+        	v -= 1;
+        	mp[u].push_back(v);
+        	mp[v].push_back(u);
+    	}
+
+    	bfs(0, 0);  // finding the distance of every node from node-0
+    	bfs(1, 1);  // finding the distance of every node from node-1
+    	bfs(N - 1, 2);  // finding the distance of every node from node-(N-1)
+
+    	int A = INT_MAX;
```

Comment:
what does `A` stand for? maybe rename it to `min_energy` or something?

Reply:
I have made all the requested changes